app/IATI/Traits/IatiValidatorResponseTrait.php

```diff
@@ -0,0 +1,441 @@
+
+        if (isset($details[0])) {
+            foreach ($error['details'] as $key => $detailedError) {
+                if (isset($detailedError['error']['line'])) {
+                    $detailWithErrorLine = $this->getDetailWithErrorLineResponse($detailedError, $activity);
+                    $error['response'][$key]['iati_path'] = $detailWithErrorLine['iati_path'];
+                    $error['response'][$key]['message'] = $detailWithErrorLine['message'];
+                }
+            }
+        } elseif (isset($details['caseContext']['paths']) && !empty($details['caseContext']['paths'])) {
+            foreach ($error['details']['caseContext']['paths'] as $caseContext => $detailedErrors) {
+                if (isset($detailedErrors['lineNumber'])) {
+                    $caseContextErrorResponse = $this->getCaseContextErrorResponse($detailedErrors, $activity, $errorMessage);
+                    $error['response'][$caseContext]['iati_path'] = $caseContextErrorResponse['iati_path'];
+                    $error['response'][$caseContext]['message'] = $caseContextErrorResponse['message'];
+                }
+            }
+        } elseif (!empty($details['caseContext']['less']) || !empty($details['caseContext']['start'])) {
+            $getCaseContextStartErrorResponse = $this->getCaseContextStartErrorResponse($error, $activity, $errorMessage);
+            $error['response'][0]['iati_path'] = $getCaseContextStartErrorResponse['iati_path'];
+            $error['response'][0]['message'] = $getCaseContextStartErrorResponse['message'];
+        } elseif (isset($details['xpathContext']['lineNumber'])) {
+            $errorResponse = $this->getXpathContextWithLineNumberErrorResponse($error, $activity, $errorMessage);
+            $error['response'][0]['iati_path'] = $errorResponse['iati_path'];
+            $error['response'][0]['message'] = $errorResponse['message'];
+        } elseif (isset($error['details']['xpath'])) {
+            $lastSegment = str_replace('//iati-activity/', '', $error['details']['xpath']);
+            $lastSegment = in_array($lastSegment, ['@budget-not-provided', '@humanitarian', '@hierarchy']) ? 'default_values' : $lastSegment;
+            $error['response'][0]['iati_path'] = checkUrlExists("/activity/$activity->id/$lastSegment") ? "/activity/$activity->id/$lastSegment" : "/activity/$activity->id";
+            $error['response'][0]['message'] = "activity>$activity->id";
+        } else {
+            $error['response'][0]['iati_path'] = "/activity/$activity->id";
+            $error['response'][0]['message'] = "activity>$activity->id";
+        }
+
+        return $error;
+    }
+
+    /**
+     * Returns Node path location.
+     *
+     * @param $nodePath
+     *
+     * @return string
+     */
+    public function getErrorLocation($nodePath): String
+    {
+        if (strpos($nodePath, 'edit')) {
+            $stringBetween = getStringBetween($nodePath, '/', '/edit');
+        } else {
```

Comment:
try to minimize the use of else if possible i.e. set $stringBetween = getStringBetween($nodePath, '/', '#'); initially and then change the value if the if statement is satisfied.